lib/api/apiUtils/object/coldStorage.js

```diff
@@ -198,9 +197,34 @@ function startRestore(objectMD, restoreParam, log, cb) {
     return cb(null, isObjectAlreadyRestored(objectMD, log));
 }
 
+/**
+ * checks if object data is available or if it's in cold storage
+ * @param {ObjectMD} objMD Object metadata
+ * @returns {ArsenalError|null} error if object data is not available
+ */
+function verifyColdObjectAvailable(objMD) {
+    // return error when object is cold
+    if (objMD.archive &&
+        // Object is in cold backend
+        (!objMD.archive.restoreRequestedAt ||
+        // Object is being restored
+        (objMD.archive.restoreRequestedAt && !objMD.archive.restoreCompletedAt))) {
```

Comment:
For my understanding, why is the existence of `objMD.archive` alone not enough to throw error? Is it maintained as an attribute on the object once a restore is successful?

Reply:
When an object is restored `objMD.archive` is still present in the metadata, and points to the data on the cold side.

Restoration is a temporary state that eventually expires, that's why we need to keep it.